fix: WASM search test assumes stale ranking order for tied scores

## Bug

The WASM test `search should return sorted results` asserts that searching for `'type'` in `['TypeScript', 'JavaScript', 'Python', 'TypeSpec']` should return `TypeScript` as the top result.

This expectation was correct when the test was written (commit `2573206`, March 14), but PR #237 (March 15) introduced a length-based tiebreaker: when scores are equal, shorter items rank first.

Since both `TypeSpec` (8 bytes) and `TypeScript` (10 bytes) score 1.0 for the query `'type'`, the tiebreaker correctly promotes `TypeSpec` over `TypeScript`.

## Impact

Local `pnpm test` fails with:
```
AssertionError: expected 'TypeSpec' to be 'TypeScript'
```

This does not affect CI because the Node.js test matrix does not include the WASM binary, so `wasm.spec.ts` is skipped there.

## Fix

Replace the fragile specific-item assertion with a sorted-order check that matches the test's stated intent ("search should return sorted results").

```diff
-      expect(results[0].item).toBe('TypeScript');
+      // Verify results are sorted by score descending
+      for (let i = 1; i < results.length; i++) {
+        expect(results[i - 1].score).toBeGreaterThanOrEqual(results[i].score);
+      }
```

--- a/__test__/wasm.spec.ts
+++ b/__test__/wasm.spec.ts
@@ -325,7 +325,10 @@ describe.skipIf(!wasmAvailable)('wasm', () => {
     it('search should return sorted results', () => {
       const results = wasm.search('type', ['TypeScript', 'JavaScript', 'Python', 'TypeSpec']);
       expect(results.length).toBeGreaterThan(0);
-      expect(results[0].item).toBe('TypeScript');
+      // Verify results are sorted by score descending
+      for (let i = 1; i < results.length; i++) {
+        expect(results[i - 1].score).toBeGreaterThanOrEqual(results[i].score);
+      }
       expect(results[0]).toHaveProperty('score');
       expect(results[0]).toHaveProperty('index');
     });

